msal/build.gradle

```diff
@@ -124,6 +134,38 @@ task sourcesJar(type: Jar) {
     destinationDir = reporting.file("$project.buildDir/outputs/jar/")
 }
 
+task createMsalCoverageReportTask() {
+    group = "Msal Reporting"
+    description = "Generate Coverage Report tasks for every build Flavour"
+    android.libraryVariants.all { variant ->
+        def buildVariant = variant.name
+        def testTaskName = "test${buildVariant.capitalize()}UnitTest"
+
+        // Tasks for generating Coverage reports for every buildVariant
+        task "${buildVariant}MsalCoverageReport" (type:JacocoReport, dependsOn: ["$testTaskName", "create${buildVariant.capitalize()}CoverageReport"]) {
```

Comment:
So tell me if my understanding is right - what we're doing here is as follows:

- For each build variant generate a new coverage report task
- This coverage report task will run unit test for the build variant specified and perform code coverage. It will also call the pre-generated jacoco task for capturing code coverage for instrumented tests. (The pre-generated task does both 
 i.e. run instrumented tests and capture coverage)

You are only using the pre-generated code coverage task to avoid having to re-do the work for doing coverage for instrumented tests?

Reply:
I think the better thing to do (as discussed the other day) is to create a separate task for coverage for unit tests. Then you can have a third task then can just combine the tasks for coverage report for unit tests + instrumented tests. This would allow us to be more flexible.

Reply:
Yes you are right, So Code coverage task is dependent on 2 tasks:

- the first one runs all the unit tests and and jacoco instruments all the src files and with the help of these two, jacoco creates final coverage report

- And second task is one which does both the task for instrument tests, so there is no need of any addition work on that task.


And as per our discussion that day, as Azure dev ops does not support instrument tests, so in all for every build there will be three tasks:

1. Task which would generate report for both unit tests and instrumented tests
2. Task which would generate report only for unit tests
3. Task which would generate report only for instrumented tests